pandas/_testing/asserters.py

```diff
@@ -265,8 +267,9 @@ def assert_index_equal(
     right : Index
     exact : bool or {'equiv'}, default 'equiv'
         Whether to check the Index class, dtype and inferred_type
-        are identical. If 'equiv', then RangeIndex can be substituted for
-        Int64Index as well.
+        are identical. If 'equiv', RangeIndex can be substituted for
+        Int64Index and integer dtypes will be equivalent to each other and
+        float dtypes equivalent to each other.
```

Comment:
Should this be done by default? A different dtype sounds as something to raise by default IMO (RangeIndex and Int64Index is something else, as RangeIndex becomes int64 when no longer a range)

Reply:
Yeah, I think you're right, will change it.